Cabal/Distribution/Compat/Prelude.hs

```diff
@@ -122,6 +123,12 @@ import Data.Maybe
 
 import qualified Text.PrettyPrint as Disp
 
+import qualified Prelude as OrigPrelude
+import Distribution.Compat.Stack
+
+type IO a = WithCallStack (OrigPrelude.IO a)
+type NoCallStackIO a = OrigPrelude.IO a
```

Comment:
This is what I'm sceptical of. Could we reverse the defaults and positively mark things instead?

Looking through this patch it's a little worrying that we may have to think carefully "should this IO helper be marked NoCallstackIO?". Or is there a clear principle here that means it's only a bunch of utility layer things and we don't have to worry about it generally?